test/tests/functions/solution_function/solution_function_grad_p2.i

```diff
@@ -0,0 +1,95 @@
+[Mesh]
+  type = GeneratedMesh
+  dim = 2
+  nx = 10
+  ny = 10
+  xmin = 0.0
+  xmax = 1.0
+  ymin = 0.0
+  ymax = 1.0
+  parallel_type = replicated
+[]
+
+[Variables]
+  [./u]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[AuxVariables]
+  [./test_variable_x]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+  [./test_variable_y]
+    order = FIRST
+    family = LAGRANGE
+  [../]
+[]
+
+[AuxKernels]
+  [./test_variable_x_aux]
+    type = FunctionGradAux
+    variable = test_variable_x
+    dimension_index = 0
+    function = solution_function
+  [../]
+  [./test_variable_y_aux]
+    type = FunctionGradAux
+    variable = test_variable_y
+    dimension_index = 1
+    function = solution_function
+  [../]
+[]
+
+[UserObjects]
+  [./ex_soln]
+    type = SolutionUserObject
+    system_variables = test_variable
+    mesh = solution_function_grad_p1.e
+  [../]
+[]
+
+[Functions]
+  [./solution_function]
+    type = SolutionFunction
+    solution = ex_soln
+  [../]
+[]
+
+[Kernels]
+  [./diff]
+    type = Diffusion
+    variable = u
+  [../]
+[]
+
+[BCs]
+  [./left]
+    type = DirichletBC
+    variable = u
+    boundary = 1
+    value = 0
+  [../]
+  [./right]
+    type = DirichletBC
+    variable = u
+    boundary = 2
+    value = 1
+  [../]
+[]
+
+[Executioner]
+  type = Steady
+
+  # Preconditioned JFNK (default)
```

Comment:
We need to stop copying this comment around! The same in the other test files...